src/ray/gcs/gcs_server/gcs_task_manager.h

```diff
@@ -128,15 +150,23 @@ class GcsTaskManager : public rpc::TaskInfoHandler {
     /// Max number of task events allowed in the storage.
     const size_t max_num_task_events_ = 0;
 
-    /// Current task events stored.
-    std::vector<rpc::TaskEvents> task_events_;
-
     /// A iterator into task_events_ that determines which element to be overwritten.
     size_t next_idx_to_overwrite_ = 0;
 
-    /// Index from task attempt to the index of the corresponding task event.
+    /// Current task events stored.
+    std::vector<rpc::TaskEvents> task_events_;
```

Comment:
I think we should really refactor this into a single LRU data structure for understanding; this can be a followup PR.

Reply:
Yep - tracking it here: https://github.com/ray-project/ray/issues/31158